core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala

```diff
@@ -583,3 +587,23 @@ private class FirehoseListenerThatAcceptsSparkConf(conf: SparkConf) extends Spar
     case _ =>
   }
 }
+
+private class SlowDeserializable(var readDuration: Int)
+  extends Externalizable {
+
+  def this() = this(0)
```

Comment:
Is this needed for deserialization? I thought it was ok to not have a default constructor.

Reply:
Nevermind, thinking of `Serializable`.

"When an Externalizable object is reconstructed, an instance is created using the public no-arg constructor"